blog/content/post/2025-05-07-dasharo-ci.md

```diff
@@ -0,0 +1,293 @@
+---
+title: 'Automating Firmware Security: CI for DBX and Microcode Updates in Dasharo'
+abstract: 'Microcode and DBX are critical components in establishing the
+          security of your platform. This blog post will discuss how Dasharo
+          automates their updates, making our firmware more transparent and
+          your platform more secure.'
+cover: /covers/dasharo-sygnet.svg
+author: michal.kopec
+layout: post
+published: true
+date: 2025-05-07
+archives: "2025"
+
+tags:
+  - coreboot
+  - uefi
+categories:
+  - Firmware
+  - Security
+
+---
+
+## Introduction
+
+From the moment you press the power button, the security of your device begins
+with your boot firmware. It plays a critical role in establishing the
+trustworthiness of your device.
+
+Dasharo is our open-source firmware distribution based on coreboot, which is
+a lightweight initial bootloader that initializes the hardware platform before
+passing control to a secondary bootloader. In Dasharo, the secondary bootloader
+is typically TianoCore EDK2, which is an implementation of UEFI.
+
+One of the responsibilities of coreboot is to provide microcode updates for the
+processor. Microcode updates are critical for firmware security, as they contain
+fixes and mitigations for many classes of exploits, like
+[Spectre and Meltdown](https://meltdownattack.com/).
+
+On the UEFI side, there is UEFI Secure Boot, which is a method for
+cryptographically verifying OS bootloaders before executing them. One of the
+components of UEFI Secure Boot is a revocation database, which is called DBX.
+It contains hashes of revoked binaries, such as bootloaders with known security
+vulnerabilities, like [GRUB and the BootHole exploit](https://eclypsium.com/blog/theres-a-hole-in-the-boot/).
+This mechanism provides a way to revoke binaries that were previously considered
+trusted.
+
+When it comes to overall firmware security trust chain, microcode sits at the
+very beginning, and UEFI Secure Boot sits at the very end, just before the OS
+starts to load. This is why updating these components regularly is crucial for
+platform security, and why we need automatic checks that ensure these components
+are up to date.
```

Comment:
Dasharo boot chain diagram would be appreciated where in platform boot uCode loading takes place and where is UEFI.

Reply:
added in https://github.com/3mdeb/news-and-ideas/pull/659/commits/dda290b4598e2c46518791ede9a67ef5e7677b68

Reply:
![](https://www.intel.com/content/dam/developer/articles/technical/software-security-guidance/resources/key-usage-ifwi/ifwi-figure2-iwfi-boot-flow-with-boot-guard.png)

![](https://igor-blue.github.io/images/image2020-4-20_14-56-23.png)

This is much better way of presenting what is going on. It comes from [here](https://www.intel.com/content/www/us/en/developer/articles/technical/software-security-guidance/resources/key-usage-in-integrated-firmware-images.html) and [here](https://igor-blue.github.io/2021/02/04/secure-boot.html).

Reply:
made horizontal in https://github.com/3mdeb/news-and-ideas/pull/659/commits/2fdf1cc2597f30e2fffd51137f740517e43d66fa